Source/JavaScriptCore/runtime/ISO8601.h

```diff
@@ -170,6 +170,43 @@ class ExactTime {
     Int128 m_epochNanoseconds { };
 };
 
+// https://tc39.es/proposal-temporal/#sec-temporal-internal-duration-records
+// Represents a duration as an ISO8601::Duration (in which all time fields
+// are ignored) along with an Int128 time duration that represents the sum
+// of all time fields. Used to avoid losing precision in intermediate calculations.
+class InternalDuration final {
+public:
+    InternalDuration(Duration d, Int128 t)
+        : m_dateDuration(d), m_time(t) { }
+    InternalDuration()
+        : m_dateDuration(Duration()), m_time(0) { }
+    static constexpr Int128 maxTimeDuration = 9'007'199'254'740'992 * ExactTime::nsPerSecond - 1;
+
+    int32_t sign() const;
```

Comment:
Should we implement this `sign` method? Maybe https://tc39.es/proposal-temporal/#sec-durationsign ?

Reply:
Done in [656e2e3](https://github.com/WebKit/WebKit/pull/43980/commits/656e2e3542816ef9c2ebdf44df642c6fa2d7d842) (it was already implemented, I just hadn't pulled it in when I was splitting up the original PR).